django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,24 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        registered_tables = {
+            model._meta.db_table: model
+            for model in self.opts.apps.get_models(include_auto_created=True)
+            if model != self.remote_field.through
+        }
+        if not isinstance(self.remote_field.through, str) and self.m2m_db_table() in set(registered_tables.keys()):
```

Comment:
I suggest you skip the check (`return []` early) if the intermediary model (`self.remote_field.through`) is not resolved.

That is `isinstance(self.remote_field.through, six.string_types)`. Also I would store `m2m_db_table` in a variable as you'll need to reuse it to lookup `registered_tables` below.